### ISSUE TITLE
Policyshard `combine` does not take conditions into account well

### ISSUE THREAD

**Issue by Sam-Martin** [CloudWanderer-io/PolicyGlass#10]
See 
```
tests/unit/difference/test_policyshard.py::deny_action_and_resource_subsets
tests/unit/difference/test_policyshard.py::exact_match_on_inclusion_with_exclusion_and_condition
tests/unit/test_policy_shards_effect.py::complex_exception
````
for an example.

it is technically _correct_ and doesn't state something is denied that's not, but simply provides a slightly confusing result because it shows the condition as being larger than it is even though there's another allow that covers some of the same ARP area.

**Comment by Sam-Martin** [comment]
I believe this was resolved in 0.4.7 #13

**Comment by Sam-Martin** [comment]
This has now been resolved with the creation of `delineate_intersecting_shards` in #16
